### ISSUE TITLE
preprocessed mimic-cxr annotation seems to be different with others

### ISSUE THREAD

**Issue by zh-Wang-Med** [wang-zhanyu/R2GenGPT#7]
your preprocessed mimic-cxr annotation contains impression and finding. However, other's setting usually only use finding section of a report, such as R2Gen model. And I run your model on annotations only containging finding, the result is lower than other's. Can you explain? I'm not sure if I made a mistake somewhere

**Comment by wang-zhanyu** [comment]
Thanks for your interest. We referred to the official preprocessing code provided for parsing the reports, which contain both impressions and findings. Considering that both sections are crucial for a complete report, we retained them entirely. Regarding the experimental phenomenon you observed, could you please provide the experimental results and specify "others"?

**Comment by zh-Wang-Med** [comment]
Thanks for your response! for example, R2Gen is a famous baseline, I notice you directly adopted the results given in their article.
However, their just predicted finding section ranther than impression and finding. Would it be unfair for you to directly compare like this? And I try your model R2GenGPT on report with only finding section, the result is lower than yours based on finding + impression.

**Comment by zh-Wang-Med** [comment]
This is the result of using your code to only predict the finding section, the CIDEr is a little lower
Test result of /home/xxx/R2GenGPT-main/save/mimic_cxr/v1_deep_finding/checkpoints/checkpoint_epoch4_step112827_bleu0.132891_cider0.201495.pth: {'Bleu_1': 0.40156787140109074, 'Bleu_2': 0.24777515566553326, 'Bleu_3': 0.16587956792259043, 'Bleu_4': 0.11786933490873241, 'ROUGE_L': 0.27735867532555053, 'METEOR': 0.15643617274462107, 'CIDEr': 0.1991703269894305}
And this is the result of using your code to predict the finding and impression section just like you, the result is consistent with the results of your paper 
Test result of /home/xxx/R2GenGPT/save/mimic_cxr/v1_test2_deep/checkpoints/checkpoint_epoch8_step135396_bleu0.190921_cider0.370094.pth: {'Bleu_1': 0.4128636987578053, 'Bleu_2': 0.2695699159478738, 'Bleu_3': 0.18779073357219533, 'Bleu_4': 0.13655461790879653, 'ROUGE_L': 0.2974765397369367, 'METEOR': 0.1622455704235942, 'CIDEr': 0.26411428019543615}

**Comment by wang-zhanyu** [comment]
Thank you for pointing out this issue.  As we have mentioned, our work processes the official dataset, which includes both the impression and findings sections. The impression is a crucial part of a diagnostic report, and we should not omit it just because an earlier work did not use it. 

Additionally, we would like to clarify that in the results reported in our paper, any method that was not fairly reproduced is marked with a dagger symbol. This is because many works do not have open-source code,  in my experience, even different data preprocessing can result in variations in outcomes, making comparisons with these methods difficult to ensure absolute fairness. You can refer to methods without the dagger symbol, these are the ones we have replicated ourselves under the same experimental setup and can be compared more fairly.

Still, it's interesting to note your point. We trained our model using only findings and achieved the following results: {'Bleu_1': 0.404, 'Bleu_2': 0.252, 'Bleu_3': 0.169, 'Bleu_4': 0.121, 'ROUGE_L': 0.277, 'METEOR': 0.155, 'CIDEr': 0.209}.  If you prefer to use only findings, you are welcome to use this result.

**Comment by passby111** [comment]
Thank&nbsp;you&nbsp;for&nbsp;your&nbsp;attention&nbsp;and&nbsp;efforts&nbsp;of&nbsp;retraining&nbsp;model&nbsp;on&nbsp;the&nbsp;finding&nbsp;section. I&nbsp;am&nbsp;willing&nbsp;to&nbsp;use&nbsp;your&nbsp;result. I agree with you about "different data preprocessing can result in variations in outcomes"&nbsp;
In&nbsp;addition, I&nbsp;have&nbsp;another&nbsp;question&nbsp;about&nbsp;your&nbsp;result&nbsp;of CE. I&nbsp;use&nbsp;https://github.com/stanfordmlgroup/chexpert-labeler to label test refs and test result, and use&nbsp;https://github.com/zhjohnchan/R2Gen/blob/main/compute_ce.py to calculate the result. The result of predicting finding and impression is&nbsp; 'F1_MACRO': 0.23245815170872425, 'F1_MICRO': 0.409047345217558, 'PRECISION_MACRO': 0.33660613529579375, 'PRECISION_MICRO': 0.4855885922330097, 'RECALL_MACRO': 0.22065983503584846, 'RECALL_MICRO': 0.35335025941053094}
The result seems&nbsp;to&nbsp;be&nbsp;wrong.&nbsp;May I ask how you implemented the calculation at that time? I would greatly appreciate your help.
Thank you !








------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "wang-zhanyu/R2GenGPT"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年2月26日(星期一) 上午10:07
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [wang-zhanyu/R2GenGPT] preprocessed mimic-cxr annotation seems to be different with others (Issue #7)





 
Thank you for pointing out this issue.  As we have mentioned, our work processes the official dataset, which includes both the impression and findings sections. The impression is a crucial part of a diagnostic report, and we should not omit it just because an earlier work did not use it.
 
Additionally, we would like to clarify that in the results reported in our paper, any method that was not fairly reproduced is marked with a dagger symbol. This is because many works do not have open-source code,  in my experience, even different data preprocessing can result in variations in outcomes, making comparisons with these methods difficult to ensure absolute fairness. You can refer to methods without the dagger symbol, these are the ones we have replicated ourselves under the same experimental setup and can be compared more fairly.
 
Still, it's interesting to note your point. We trained our model using only findings and achieved the following results: {'Bleu_1': 0.404, 'Bleu_2': 0.252, 'Bleu_3': 0.169, 'Bleu_4': 0.121, 'ROUGE_L': 0.277, 'METEOR': 0.155, 'CIDEr': 0.209}.  If you prefer to use only findings, you are welcome to use this result.

**Comment by zh-Wang-Med** [comment]
Thank you for your attention and efforts of retraining model on the finding section. I am willing to use your result. I agree with you about "different data preprocessing can result in variations in outcomes"
In addition, I have another question about your result of CE. I use https://github.com/stanfordmlgroup/chexpert-labeler to label test refs and test result, and use https://github.com/zhjohnchan/R2Gen/blob/main/compute_ce.py to calculate the result. The result of predicting finding and impression is  'F1_MACRO': 0.23245815170872425, 'F1_MICRO': 0.409047345217558, 'PRECISION_MACRO': 0.33660613529579375, 'PRECISION_MICRO': 0.4855885922330097, 'RECALL_MACRO': 0.22065983503584846, 'RECALL_MICRO': 0.35335025941053094}
The result seems to be wrong. May I ask how you implemented the calculation at that time? I would greatly appreciate your help.
Thank you !

**Comment by zh-Wang-Med** [comment]
hi, can you give your results of CE about your model only predicting the finding section? I think this result is also different with your results in your paper, which predict both impression and finding. Thanks.

**Comment by Stella-xixi** [comment]
> 这是使用你的代码只预测发现部分的结果，CIDEr 稍微低一点 /home/xxx/R2GenGPT-main/save/mimic_cxr/v1_deep_finding/checkpoints/checkpoint_epoch4_step112827_bleu0.132891_cider0.201495.pth 的测试结果：{'Bleu_1': 0.40156787140109074, 'Bleu_2': 0.24777515566553326, 'Bleu_3': 0.16587956792259043, 'Bleu_4': 0.11786933490873241, 'ROUGE_L': 0.27735867532555053, 'METEOR': 0.15643617274462107, 'CIDEr': 0.1991703269894305} 而这个就是像你一样用你的代码预测finding和impression部分的结果，结果和你的论文结果是一致的 /home/xxx/R2GenGPT/save/mimic_cxr/v1_test2_deep/checkpoints/checkpoint_epoch8_step135396_bleu0.190921_cider0.370094.pth的测试结果：{'Bleu_1': 0.4128636987578053, 'Bleu_2': 0.2695699159478738, 'Bleu_3'：0.18779073357219533，'Bleu_4'：0.13655461790879653，'ROUGE_L'：0.2974765397369367，'流星'：0.1622455704235942， “苹果酒”：0.26411428019543615}

Hello, I notice that you have successfully reproduced the experimental results from the original paper. However, the results I obtained when reproducing the experiment on MIMIC are significantly lower than those reported in the paper. I would like to ask how you managed to reproduce the experimental data? Specifically, what were the hyperparameter settings for the experiment, and on which server was it run?Regarding the experimental data and other related information, I'm really looking forward to your reply. Thank you.

**Comment by zh-Wang-Med** [comment]
default setting, A6000 and A800, respectively

**Comment by Stella-xixi** [comment]
您好！

非常感谢您此前的及时回复，您的分享为我解决问题提供了很大帮助，在此深表谢意。

目前我在运行相关代码时，因硬件条件所限，仅使用了一张 NVIDIA RTX A6000 显卡，同时将 batchsize 设置为 8。出于对实验细节的探究，想向您请教：您当时运行相同代码时，使用了几张 NVIDIA RTX A6000 显卡呢？

另外，我在代码运行过程中开启了 “low_resource” 模式，不确定该设置是否会对实验结果产生影响，也想听听您的经验和看法。

若您方便，盼复为感！

此致
敬礼！



------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "wang-zhanyu/R2GenGPT"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年9月16日(星期二) 中午11:25
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [wang-zhanyu/R2GenGPT] preprocessed mimic-cxr annotation seems to be different with others (Issue #7)



zh-Wang-Med left a comment (wang-zhanyu/R2GenGPT#7)
 
default setting, A6000 and A800, respectively

**Comment by zh-Wang-Med** [comment]
1张也可以，但会稍微慢些，不建议开启low_resource，会降低精度

**Comment by Stella-xixi** [comment]
Hello, Thank you very much for your reply. With your help, I've achieved significant improvements in the validation set results. However, the BLEU-4 score on the test set still dropped to 0.10, which has been quite puzzling to me.  I've kept the parameter settings at their defaults, with the only modification being the batch size. For both the vision_model and llama_model, I downloaded the weights locally.  I wonder if you could suggest other possible reasons that might cause the performance drop on the test set? Below are my training and testing parameters:
training parameters:
{'accelerator': 'gpu',
 'accumulate_grad_batches': 1,
 'annotation': '/data1/data/mimic_annotation_all.json',
 'base_dir': '/data1/Data/mimic_cxr/images',
 'batch_size': 8,
 'beam_size': 3,
 'ckpt_file': None,
 'dataset': 'mimic_cxr',
 'delta_file': None,
 'devices': 1,
 'diversity_penalty': 0,
 'do_sample': False,
 'end_sym': '</s&gt;',
 'every_n_train_steps': 0,
 'freeze_vm': False,
 'global_only': False,
 'gradient_clip_val': None,
 'learning_rate': 0.0001,
 'length_penalty': 2.0,
 'limit_test_batches': 1.0,
 'limit_train_batches': 1.0,
 'limit_val_batches': 0.5,
 'llama_model': '/data1/LLM/R2GenGPT/ckpoint/Llama-2-7b-chat-hf',
 'llm_alpha': 16,
 'llm_r': 16,
 'llm_use_lora': False,
 'lora_dropout': 0.1,
 'low_resource': False,
 'max_epochs': 5,
 'max_length': 100,
 'max_new_tokens': 120,
 'min_new_tokens': 80,
 'no_repeat_ngram_size': 2,
 'num_beam_groups': 1,
 'num_nodes': 1,
 'num_sanity_val_steps': 2,
 'num_workers': 12,
 'precision': 'bf16-mixed',
 'prefetch_factor': 4,
 'repetition_penalty': 2.0,
 'savedmodel_path': './save/mimic_cxr/v1_deep',
 'scorer_types': ['Bleu_4', 'CIDEr'],
 'strategy': 'ddp',
 'temperature': 0,
 'test': False,
 'test_batch_size': 16,
 'val_batch_size': 12,
 'val_check_interval': 0.5,
 'validate': False,
 'vis_alpha': 16,
 'vis_r': 16,
 'vis_use_lora': False,
 'vision_model': '/data1/LLM/R2GenGPT/ckpoint/swin-base-patch4-window7-224',
 'weights': [0.5, 0.5]}
Global seed set to 42testing parameters：{'accelerator': 'gpu',
 'accumulate_grad_batches': 1,
 'annotation': '/data1/data/mimic_annotation_all.json',
 'base_dir': '/data1/Data/mimic_cxr/images', 'batch_size': 6,
 'beam_size': 3,
 'ckpt_file': None,
 'dataset': 'mimic_cxr',
 'delta_file': '/data1/code/r2gengpt/save/mimic_cxr/v1_deep/checkpoints/checkpoint_epoch0_step33848_bleu0.169259_cider0.253519.pth',
 'devices': 1,
 'diversity_penalty': 0,
 'do_sample': False,
 'end_sym': '</s&gt;',
 'every_n_train_steps': 0,
 'freeze_vm': False,
 'global_only': False,
 'gradient_clip_val': None,
 'learning_rate': 0.0001,
 'length_penalty': 2.0,
 'limit_test_batches': 1.0,
 'limit_train_batches': 1.0,
 'limit_val_batches': 1.0,
  'llama_model': '/data1/LLM/R2GenGPT/ckpoint/Llama-2-7b-chat-hf',
 'llm_alpha': 16,
 'llm_r': 16,
 'llm_use_lora': False,
 'lora_dropout': 0.1,
 'low_resource': False,
 'max_epochs': 3,
 'max_length': 100,
 'max_new_tokens': 120,
 'min_new_tokens': 80,
 'no_repeat_ngram_size': 2,
 'num_beam_groups': 1,
 'num_nodes': 1,
 'num_sanity_val_steps': 2,
 'num_workers': 12,
 'precision': 'bf16-mixed',
 'prefetch_factor': 4,
 'repetition_penalty': 2.0,
 'savedmodel_path': './save/mimic_cxr/v1_deep',
 'scorer_types': ['Bleu_4', 'CIDEr'],
 'strategy': 'ddp',
 'temperature': 0,
 'test': True,
 'test_batch_size': 16,
 'val_batch_size': 16,
 'val_check_interval': 1.0,
 'validate': False,
 'vis_alpha': 16,
 'vis_r': 16,
 'vis_use_lora': False,
 'vision_model': '/data1/LLM/R2GenGPT/ckpoint/swin-base-patch4-window7-224', 'weights': [0.5, 0.5]}
Global seed set to 42


I eagerly look forward to your reply! Thank you very much!


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "wang-zhanyu/R2GenGPT"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年9月16日(星期二) 中午11:43
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [wang-zhanyu/R2GenGPT] preprocessed mimic-cxr annotation seems to be different with others (Issue #7)



zh-Wang-Med left a comment (wang-zhanyu/R2GenGPT#7)
 
1张也可以，但会稍微慢些，不建议开启low_resource，会降低精度
